plugins/decoder.rst

```diff
@@ -1,4 +1,34 @@
 Decoder Plugin
 ==============
 
-TODO
+Decoder Plugin decodes the body of the response with filters coming from the `Transfer-Encoding` or `Content-Encoding`
+headers::
+
+    use Http\Discovery\HttpClientDiscovery;
+    use Http\Client\Plugin\PluginClient;
+    use Http\Client\Plugin\DecoderPlugin;
+
+    $decoderPlugin = new DecoderPlugin();
+
+    $pluginClient = new PluginClient(
+        HttpClientDiscovery::find(),
+        [$decoderPlugin]
+    );
+
+Actually it can decodes 4 different filters:
```

Comment:
or maybe "The plugin can handle the following encodings:"

(it does not decode filters, it decodes encodings with the help of filters. the filters however are an implementation detail. thus i would speak about handling some encodings)